### PR TITLE
feat: remove nonce management, use starknet  nonce

### LINKED ISSUE #648: feat: remove nonce management from contracts, and rely on starknet nonce
# Feature Request

We as of now are [managing nonce](https://github.com/kkrt-labs/kakarot/blob/dcc93a78d9126c0101af739d6aebf4476514a62e/src/kakarot/accounts/library.cairo#L25) ourselves for EOAs and Contracts, this causes a lot of issues, for example if you flood the mempool of the sequencer with a lot of transactions, only 1 of them will pass and all following will fail on validation, as we check the nonce for validation.

**Describe Preferred Solution**

We have to remove management of nonce, and rely on starknet nonce which we should be able to obtain via [`get_tx_info`](https://docs.starknet.io/documentation/architecture_and_concepts/Contracts/system-calls-cairo0/#get_transaction_info) syscall. 

**If the feature request is approved, would you be willing to submit a PR?**
Yes

### PR DESCRIPTION
**Pull request by bajpai244**
This pr removes nonce management and uses starknet nonce wherever nonce is required.

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

We currently manage nonce ourselves which causes multiple issues, migrating to starknet nonce is a better solution and the PR implements the required changes for that.

Resolves #648 

## What is the new behavior?

- nonce management has been removed
- we now use starknet nonce wherever required

## Other information

- The rpc is already [using starknet nonce](https://github.com/kkrt-labs/kakarot-rpc/blob/d6e3676de6fc9e804dc751aa3493810c95957bba/crates/core/src/client/mod.rs#L338), which means that the current transactions by tools like metamask are already being signed with starknet nonce and not our own nonce when RPC is being used.

### CODE REVIEW

**Review by ClementWalter — DISMISSED** [review]

**Review by ClementWalter — APPROVED** [review]

File: tests/src/kakarot/instructions/test_system_operations.py [hunk 1]
```diff
@@ -87,7 +87,6 @@ async def test_call__should_transfer_value(self, system_operations, mint):
             system_operations.contract_address
         )
 
-    @pytest.mark.parametrize("salt", [0, 127, 256, 2**55 - 1])
     async def test_create(self, system_operations, salt):
```
Comment (ClementWalter) [inline tests/src/kakarot/instructions/test_system_operations.py:90]: why don't you keep the `salt`?

Comment (ClementWalter) [reply tests/src/kakarot/instructions/test_system_operations.py:90]: and it's become undefined in the body I think

**Review by ClementWalter — COMMENTED** [review]
If you feel so, you can create fixtures to name used addresses once for all, and avoid magic numbers, you can put them at the top of the file

File: tests/src/kakarot/instructions/test_system_operations.py (hunk 1 above)
Comment (bajpai244) [reply tests/src/kakarot/instructions/test_system_operations.py:90]: @ClementWalter I can keep this parameterized as well, for now I just replaced the salt with 0, but I guess I can bump it back to being parameterized if you want.

More about the issue can be found here -> https://github.com/kkrt-labs/kakarot-rpc/blob/b9fcd0c574ef4aa06dd41dbc1150903154ff6a7a/crates/core/src/client/mod.rs#L412

Note: as mentioned in the above issues, @jobez will later on refactor these tests after this PR merge.

**Comment by bajpai244** [comment]
> If you feel so, you can create fixtures to name used addresses once for all, and avoid magic numbers, you can put them at the top of the file

Makes sense, lemme do that!

File: tests/src/kakarot/instructions/test_system_operations.py (hunk 1 above)
Comment (ClementWalter) [reply tests/src/kakarot/instructions/test_system_operations.py:90]: just need to remove the arg I guess.

**Review by ClementWalter — DISMISSED** [review]

**Review by ClementWalter — APPROVED** [review]

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**